anndata/_config.py

```diff
@@ -288,11 +290,77 @@ def __doc__(self):
         )
 
 
+def check_and_get_environ_var(
+    key: str,
+    default_value: str,
+    allowed_values: Sequence[str] | None = None,
+    cast: Callable[[Any], T] = lambda x: x,
+) -> T:
+    """Get the environment variable and return it is a (potentially) non-string, usable value.
+
+    Parameters
+    ----------
+    key
+        The environment variable name.
+    default_value
+        The default value for `os.environ.get`.
+    allowed_values
+        Allowable string values., by default None
+    cast
+        Casting from the string to a (potentially different) python object, by default lambdax:x
+
+    Returns
+    -------
+    The casted value.
+    """
+    environ_val = os.environ.get(key, default_value)
+    if allowed_values is not None and environ_val not in allowed_values:
+        warnings.warn(
+            f'Value "{environ_val}" is not in allowed {allowed_values} for environment variable {key}.\
+                      Default {default_value} will be used.'
+        )
+        return cast(default_value)
+    return cast(environ_val)
+
+
+def check_and_get_bool(option, default_value):
+    return check_and_get_environ_var(
+        "ANNDATA_" + option.upper(),
+        str(int(default_value)),
+        ["0", "1"],
```

Comment:
There’s a lot of conventions for this. E.g.

1. In older versions of Python, there was a function for one: https://docs.python.org/3.8/distutils/apiref.html#distutils.util.strtobool
2. https://force-color.org/ defines “true” as “present and not an empty string”

So maybe we should skip `yes`/`no` and `on`/`off`, since `no` and `on` are so close, but support all capitalizations of `True`/`False` and the empty string. What do you think?